### PR TITLE
[trainer] feat: add TensorBoard logging backend

### PR DESCRIPTION
**Pull request by Eisenhower**
## Summary

Add a lightweight TensorBoard logging callback (`TensorBoardTraceCallback`) that writes scalar training metrics (loss, lr, MFU, throughput, etc.) to event files via `torch.utils.tensorboard.SummaryWriter`.

- **Independent from wandb** — both backends can run simultaneously
- **Rank-0 only** — no file contention in distributed training
- **Graceful fallback** — logs a clear warning when `tensorboard` is not installed
- **New config**: `train.tensorboard.enable` / `train.tensorboard.save_dir`

## Motivation

Many teams prefer TensorBoard over wandb for metric visualization — it's self-hosted, has no external service dependency, and integrates natively with PyTorch. This PR adds first-class TensorBoard support without removing or changing the existing wandb integration.

## Changes

| File | Change |
| --- | --- |
| `veomni/arguments/arguments_types.py` | New `TensorBoardConfig` dataclass |
| `veomni/trainer/callbacks/trace_callback.py` | New `TensorBoardTraceCallback` class |
| `veomni/trainer/callbacks/__init__.py` | Export new callback |
| `veomni/trainer/base.py` | Integrate callback into all lifecycle hooks |
| `pyproject.toml` | Add `tensorboard` dependency |
| `docs/usage/arguments.md` | Document `TensorBoardConfig` |
| `docs/usage/trainer.md` | List new callback |
| `tests/test_tensorboard_callback.py` | 7 unit tests covering all code paths |

## Usage

```yaml
train:
  tensorboard:
    enable: true
    save_dir: /path/to/tb_logs  # optional, defaults to <output_dir>/tensorboard
```

Then visualize with:
```bash
tensorboard --logdir /path/to/tb_logs
```

## Test plan

- [x] `ruff check` + `ruff format --check` pass
- [x] 7 unit tests in `tests/test_tensorboard_callback.py` (all pass):
  - Writer creation on rank 0
  - Scalar metric writing and event file generation
  - Disabled on non-rank-0
  - Disabled when `enable=False`
  - Default `save_dir` fallback to `<output_dir>/tensorboard`
  - Graceful handling when tensorboard package is missing
  - Config dataclass correctness

### CODE REVIEW

**Comment by Eisenhower** [comment]
Hi @TimYangst @Luosuu @FoolPlayer @Coach257 — friendly ping on this one! 🙂

Just wanted to check in: is TensorBoard logging support something the team would find useful? If this PR aligns with the project's direction, I'd love to get your thoughts on next steps — happy to iterate on any feedback or address review comments promptly.

No pressure at all — if this isn't a priority or doesn't fit the current roadmap, just let me know and I can close it. Thanks for your time!

**Comment by Luosuu** [comment]
Hi @Eisenhower Thank you for your contribution. However, we do not plan to support tensorboard right now as we find wandb can cover all the useful functionalities

**Comment by Eisenhower** [comment]
Thanks for the clear guidance. That makes sense. Since TensorBoard support is not part of the current project direction and wandb already covers the needed functionality, I'll close this PR to keep the queue clean. Thanks again for taking the time to review it.
